### PR TITLE
test(core): pin vectorized equivalence to the serial NumPy API (#930)

### PR DESCRIPTION
**Pull request by monocongo**
## What

Adds the numerical-equivalence regression guard #930 asks for, and records the
tolerance it asserts.

The vectorization work already landed per-index contract tests in
`tests/test_spatial_kernel.py` (shape, mask, NaN, kernel-invocation count), but
their equivalence assertions were all `atol=1e-8` — loose enough that the actual
agreement was unstated and the divergence bound undocumented.

## Measured behavior, now asserted

Comparing the Spatial Block path against the serial NumPy API applied cell by
cell over the 5 x 6 reference grid, plus Dask-block against in-memory:

| Path | Agreement |
| --- | --- |
| SPI (gamma, pearson), SPEI, EDDI, percentage of normal | bit-for-bit (`assert_array_equal`) |
| Thornthwaite PET | 1.14e-13 max abs diff (4 float64 ULP at the worst cell), asserted at `atol=1e-12` |
| Dask chunk layout vs in-memory | bit-for-bit for SPI, EDDI, percentage of normal; same PET tolerance |

SPEI is exercised with a per-cell-varying PET field, so a block path that pairs
a cell with the wrong PET series fails the test (rolling the field moves the
result by 1.46); the Thornthwaite reference calls `indices.pet` with an explicit
`data_start_year` rather than the adapter's own per-cell path.

`docs/performance.md` gains a `Numerical equivalence` section stating the policy:
anything beyond these bounds means the block path changed the science, and it
fails `pytest` instead of silently moving an index value. The claim is scoped to
fully populated grids, and names the whole-block Pearson fallback (ADR-0009) and
the missing-data/partial-year/daily shapes that keep their looser bounds.

## Scope decisions

- The existing `atol=1e-8` assertions in `tests/test_spatial_kernel.py` are left
  as they are. They pass, they cover per-index contracts the new module does not
  repeat (NaN, partial final year, daily grids), and rewriting them is unrelated
  churn with cross-platform flake risk; the new module is the tight bound.
- Hargreaves PET is not in the new module: its grid fixture is daily and local to
  `tests/test_spatial_kernel.py`, which already pins it at 1e-8. #930's "SPI and
  at least one other index" is met by SPI plus four more.
- The Palmer block path already asserts bit-for-bit in
  `tests/test_palmer_spatial.py`; nothing added there.
- `CHANGELOG.md` untouched: `docs/release-process.md` updates it in the release
  PR, and merged PRs in this area do not touch it.

## Verification

```
uv run ruff check src/ tests/          All checks passed!
uv run ruff format --check src/ tests/ 95 files already formatted
uv run mypy src/                       Success: no issues found in 28 source files
uv run pytest                          2151 passed, 423 deselected
uv run scripts/generate_llms_txt.py    llms-full.txt regenerated (docs/performance.md is a source)
```

The bounds were checked for discriminating power: rolling the SPEI PET field
fails the SPEI test (max diff 1.46), and dropping the Thornthwaite tolerance to
`1e-15` fails the PET test.

Refs #930

## Summary by Sourcery

Pin numerical equivalence between spatial index calculations and their serial NumPy references, with documented tolerances for known floating-point differences.

Enhancements:
- Define and enforce numerical-equivalence bounds between spatial, serial NumPy, and Dask-backed index calculations.
- Document the numerical-equivalence policy, supported tolerances, and intentional block-level Pearson fallback semantics.

Documentation:
- Add user-facing documentation describing exact and tolerance-based equivalence expectations across spatial and chunked calculations.

Tests:
- Add regression coverage for SPI, SPEI, EDDI, percentage of normal, Thornthwaite PET, and Dask chunk-layout equivalence against serial or in-memory results.

Chores:
- Regenerate the full LLM documentation output to include the performance documentation changes.

## Summary by CodeRabbit

- **Documentation**
  - Documented numerical equivalence between spatial calculations and serial NumPy results.
  - Clarified precision guarantees for SPI, SPEI, EDDI, percentage of normal, PET, and Pearson fitting.
  - Documented behavior for chunked data, fallback processing, daily calendars, and incomplete data.

- **Tests**
  - Added comprehensive numerical-equivalence coverage across supported indices, distributions, and Dask-backed grids.
  - Verified expected tolerances, bit-for-bit results where applicable, and preservation of lazy chunking.

### CODE REVIEW

File: docs/performance.md [hunk 1]
```diff
@@ -91,6 +91,21 @@ same chunking on the dimensions they share with the other input: each input's
 `time` chunking is validated independently, and a mismatch on `lat`/`lon`
 survives into the compute as a Dask `rechunk-merge` copy.
 
+## Numerical equivalence
+
+The block kernel runs the same NumPy core as the serial API, so a gridded result is
```
[the remark this answers is not in the document]
Comment (monocongo) [reply docs/performance.md]: Addressed in `1d4a0845` and `4422807b` (reviewed at `dc823b84`). The policy is now scoped to monthly grids: "on a fully populated monthly grid a gridded result is the serial result". The daily path is stated explicitly — daily xarray grids are calendar-adapted to an all-leap 366-day series before the NumPy core and converted back afterward, so their serial counterpart is that adapted series, not a raw Gregorian NumPy input — and daily-grid shapes are pointed at the looser bounds in `tests/test_spatial_kernel.py` (`atol=1e-8`, `rtol=1e-7`).

File: tests/test_numerical_equivalence.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+import numpy as np
+import pytest
+import xarray as xr
+
+from climate_indices import compute, eddi, indices, percentage_of_normal, pet_thornthwaite, spei, spi
+
+# The only path whose block form reorders arithmetic: the measured spread is 5.7e-14
+# (about one float64 ULP) on the reference grid, asserted here with four orders of
+# headroom so the bound still trips on any change that moves a cell's value.
```
[the remark this answers is not in the document]
Comment (monocongo) [reply tests/test_numerical_equivalence.py]: Addressed in `1d4a0845` and `4422807b`. The quoted spread now comes from this module\u0027s 5 x 6 fixture — 1.14e-13, about 4 float64 ULP at the worst cell, roughly an order of magnitude of headroom under `atol=1e-12` — and the comment no longer claims one-ULP agreement. The old 5.7e-14 came from the 3 x 2 fixture in `tests/test_spatial_kernel.py`. The final clause now states the allowance directly: the bound trips on regressions larger than itself, drift below the bound is accepted, not reported.

File: docs/performance.md [hunk 3]
```diff
@@ -91,6 +91,21 @@ same chunking on the dimensions they share with the other input: each input's
...
+the serial result: SPI, SPEI, EDDI, and percentage of normal are bit-for-bit
+identical to calling the NumPy API once per cell. Thornthwaite PET is the one
+exception — its block form reorders the same arithmetic and differs by a few float64
+ULP (5.7e-14 measured on the reference grid), asserted at `atol=1e-12`.
+
+Chunk layout does not change the numbers either: the same grid computed from a
+different chunk shape matches the in-memory result bit for bit, within the same
+Thornthwaite tolerance. The bounds are enforced by
+`tests/test_numerical_equivalence.py` for SPI, SPEI, EDDI, percentage of normal, and
```
[the remark this answers is not in the document]
Comment (monocongo) [reply docs/performance.md]: Addressed in `1d4a0845`. The chunk-layout claim now names only the paths the Dask comparison covers: bit-for-bit for SPI, EDDI, and percentage of normal, and within the Thornthwaite tolerance for PET; SPEI is no longer named. The Dask test gained EDDI and percentage of normal, and the PET case rechunks to `{"lat": 2, "lon": 3}` so blocks actually recombine. The Pearson case is documented as Spatial Block semantics (whole-block gamma fallback, ADR-0009) rather than per-cell equivalence, so the uncovered branch is no longer claimed.

File: tests/test_numerical_equivalence.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+def test_thornthwaite_block_path_stays_within_the_documented_tolerance(
+    gridded_monthly_temp_3d: xr.DataArray,
+) -> None:
+    """Thornthwaite PET over a Spatial Block agrees with the per-cell result within one ULP."""
+    latitudes = gridded_monthly_temp_3d.coords["lat"].values
+
+    result = pet_thornthwaite(gridded_monthly_temp_3d, xr.DataArray(latitudes, dims=["lat"]))
+    expected = _pointwise(
+        gridded_monthly_temp_3d,
+        lambda series, latitude: pet_thornthwaite(series, latitude).values,
```
[the remark this answers is not in the document]
Comment (monocongo) [reply tests/test_numerical_equivalence.py]: Addressed in `1d4a0845`. The Thornthwaite reference now calls the NumPy API directly — `indices.pet(np.asarray(series), latitude, data_start_year)` with the start year read from the grid\u0027s time coordinate — matching the other four references and no longer routing through the xarray adapter, so adapter routing and time inference are no longer shared with the result under test.

**Comment by monocongo** [comment]
### 🛠️ Multi-Agent Review Summary
- **Review target:** https://github.com/monocongo/climate_indices/pull/1026 (`test/930-vectorized-equivalence`, head `1d4a0845`)
- **Lanes run:** 3 lanes returned — tests (deepseek-flash), docs (deepseek-flash), contracts/maintainability (deepseek-flash). `lane-docs-codex` and `lane-contracts-codex` failed first on a provider quota (Codex usage limit) and were re-dispatched on DeepSeek per the provider-retry rule.
- **Validation:** `uv run ruff check src/ tests/` pass; `uv run ruff format --check src/ tests/` pass; `uv run mypy src/` pass; `uv run pytest` 2151 passed, 423 deselected; `uv run scripts/generate_llms_txt.py` leaves `llms.txt`/`llms-full.txt` byte-identical to the commit. Discriminating-power checks: rolling the SPEI PET field moves the result by 1.46 and fails the test; `atol=1e-15` on Thornthwaite fails.
- **Fix commits:** [1d4a0845 — test(core): strengthen the equivalence guard after review (#930)](https://github.com/monocongo/climate_indices/commit/1d4a0845)

### Findings and fix status

#### [P2] SPEI equivalence used a constant PET field, blind to per-cell PET misassignment — Fixed
- **Location:** `tests/test_numerical_equivalence.py` (SPEI test, lines 84–124)
- **Origin:** Tests
- **Problem:** A constant PET grid is invariant under every per-cell permutation, so a block path that paired a cell with another cell's PET series would still have passed — the one regression class the vectorized path can realistically introduce. Demonstrated by rolling the PET field and observing a 0.0 difference in the tested result versus 2.56 with a varying field.
- **Applied fix:** PET is now `precip * 0.25 + 30.0` and `_pointwise` gained an optional secondary grid so each cell's kernel receives that cell's PET series. Verified: rolling the PET field now changes the result by 1.46 and the assertion fails. Commit `1d4a0845`.

#### [P2] Published Thornthwaite spread and headroom were wrong for the asserted fixture — Fixed
- **Location:** `docs/performance.md:100`, `llms-full.txt`, `tests/test_numerical_equivalence.py:22–28`
- **Origin:** Tests, Docs, Maintainability
- **Problem:** The prose and comment quoted 5.7e-14 and "four orders of headroom"; that value comes from the 3 x 2 `gridded_monthly_temps` fixture in `tests/test_spatial_kernel.py:1000`, not the 5 x 6 `gridded_monthly_temp_3d` fixture this module asserts on, which measures 1.137e-13 (4 float64 ULP at the worst cell, value 145.76). `atol=1e-12` therefore leaves ~8.8x headroom. A reader trusting the doc would tighten the bound to 1e-13 and break the suite.
- **Applied fix:** Docs, generated bundle, and the comment now state 1.14e-13 / a few ULP / about one order of headroom. Commit `1d4a0845`.

#### [P2] "Bit-for-bit" doc claim was unconditional while block semantics override it — Fixed
- **Location:** `docs/performance.md:96–107`
- **Origin:** Docs
- **Problem:** The claim covered SPI without qualification, but `indices.spi` enables `fallback_to_gamma=True`, so a failed Pearson fit re-fits the whole block as gamma (ADR-0009, pinned by `tests/test_spatial_kernel.py:621`) and a block's cells are answered from one fit. On a block with missing cells the block result and the per-cell result diverge by design, so the unqualified claim is false.
- **Applied fix:** The section is scoped to fully populated grids and states the whole-block fallback with a link to ADR-0009. Commit `1d4a0845`.

#### [P3] Thornthwaite's reference was the adapter's per-cell path, sharing inference with the block path under test — Fixed
- **Location:** `tests/test_numerical_equivalence.py` (Thornthwaite test)
- **Origin:** Tests
- **Problem:** The reference called `pet_thornthwaite`, i.e. the same adapter layer, so both sides shared `_infer_data_start_year`; monkeypatching that inference to a wrong year left the test green while the block output diverged from a true NumPy reference by 2.14.
- **Applied fix:** The reference now calls `indices.pet(np.asarray(series), latitude, data_start_year)` with an explicit year, matching the other four tests. Commit `1d4a0845`.

#### [P3] Module docstring's ownership boundary was inaccurate — Fixed
- **Location:** `tests/test_numerical_equivalence.py:9–16`
- **Origin:** Maintainability
- **Problem:** The docstring pointed at `tests/test_spatial_kernel.py` for "shape, mask, and NaN contracts" while claiming the tight bounds were the tripwire; that file also asserts the same block-vs-reference equivalence at `atol=1e-8` (lines 311–380, 436–474, 550–553, 589, 618, 1448–1452, 1535–1550, 1612–1615, 1091–1110), so a 3e-9 drift on a NaN or partial-year grid passes both files.
- **Applied fix:** The docstring now states which shapes the tight bounds cover (fully populated grids) and that missing-data, partial-year, and daily shapes keep their `atol=1e-8` contracts in `tests/test_spatial_kernel.py`. Commit `1d4a0845`.

#### [P3] Chunk-layout claim named five indices; the test covered two, one of them with a single spatial chunk — Fixed
- **Location:** `tests/test_numerical_equivalence.py` (dask test), `docs/performance.md:102–107`
- **Origin:** Docs, Maintainability
- **Problem:** The prose claimed chunk-layout invariance for SPI, SPEI, EDDI, percentage of normal, and PET, but only SPI and Thornthwaite were compared, and `dask_monthly_temp_3d` (tests/conftest.py:751) chunks `{"lat": 5, "lon": 6}` — one spatial block, so no different layout was exercised for PET at all.
- **Applied fix:** EDDI and percentage of normal now have Dask-vs-eager comparisons, and the PET case uses `dask_monthly_temp_3d.chunk({"lat": 2, "lon": 3})`. The prose names the three bit-for-bit paths and the PET tolerance. Commit `1d4a0845`.

#### [P4] Terminology drifted from the project vocabulary — Fixed
- **Location:** `docs/performance.md`, `tests/test_numerical_equivalence.py`
- **Origin:** Docs
- **Problem:** New prose used "block kernel"/"block form" where `src/climate_indices/CONTEXT.md` defines Spatial Block, Spatial Kernel, and Per-Cell Path.
- **Applied fix:** Both files use the defined terms. Commit `1d4a0845`.

#### [P4] Commit scope `test(core)` does not match repo area scopes — Blocked
- **Location:** commit `dc823b84`
- **Origin:** Maintainability
- **Problem:** The commit also changes `docs/performance.md` and `llms-full.txt`; recent merges use area scopes (`perf(palmer)`, `docs(performance)`, `refactor(validation)`).
- **Applied fix:** **Blocked:** renaming a pushed commit requires a force-push, which is not authorized. Both commits are appended to by `1d4a0845`, so the message is a cosmetic findability issue only; squash-merge message can use `test(spatial)`.

#### [P4] Dask precip fixture is coupled to the eager one by duplicated RNG setup — Blocked
- **Location:** `tests/conftest.py:500–546` (outside this diff)
- **Origin:** Tests
- **Problem:** `dask_monthly_precip_3d` and `gridded_monthly_precip_3d` both build values with `default_rng(99)`; if one seed or distribution is later edited, the Dask comparison fails on different data rather than on chunking.
- **Applied fix:** **Blocked:** the fixtures are pre-existing and outside this diff; the failure mode is a loud mismatch, not a silent pass. Follow-up: derive the Dask fixture via `.chunk(...)` of the eager one.

### Remaining blockers
None. No P0/P1 findings. The two P4 items are unfixed as recorded above: one needs an authorized force-push, the other lives in `tests/conftest.py` outside this diff.

**Comment by monocongo** [comment]
### 🛠️ Multi-Agent Review Summary
- **Review target:** [#1026](https://github.com/monocongo/climate_indices/pull/1026) (`test/930-vectorized-equivalence`, reviewed at head `33ace2a4`, merged by the maintainer as [`a16a2d6c`](https://github.com/monocongo/climate_indices/commit/a16a2d6cc96dc134b4a8a3537fe8426f3400d432) during the fix pass)
- **Lanes run:** 4 lanes — `lane-tests-ds` (tests), `lane-correctness-ds` (numerical correctness/edge cases), `lane-docs-codex` (docs/generated bundles), `lane-contracts-codex` (API contracts/maintainability). The two codex-pinned lanes first failed on an openai-codex usage limit and were retried successfully on deepseek/deepseek-flash. A verification pass then re-ran `lane-tests-ds` and `lane-docs-codex` on the fixed head `2fb62a99`.
- **Validation:** `uv run ruff check src/ tests/` pass; `uv run ruff format --check src/ tests/` pass; `uv run mypy src/` pass; `uv run pytest` 2183 passed / 423 deselected; `uv run pytest -m validation` 378 passed / 3 skipped; CI run [35298034598](https://github.com/monocongo/climate_indices/actions/runs/35298034598) success on `2c128047`; follow-up PR [#1042](https://github.com/monocongo/climate_indices/pull/1042) CI 19 checks pass and SonarCloud gate OK (`new_duplicated_lines_density` 0.0%).
- **Fix commits:** [`046708ac` — test(core): harden the equivalence suite against fixture mutation and vacuous passes](https://github.com/monocongo/climate_indices/commit/046708acf09358a81e3a325075ab3a7b43f4a6e3), [`2fb62a99` — docs: scope numerical-equivalence claims to the enforced paths](https://github.com/monocongo/climate_indices/commit/2fb62a993258706b62a1cafd8fc1e25bfb436716), [`2c128047` — test(core): assert Pearson SPEI at the documented ULP bound](https://github.com/monocongo/climate_indices/commit/2c1280473a1d4a6b6bb229a3c0d794bc0deb587d), [`894bb803` — test(core): share the SPI equivalence helper to cut duplicated test lines](https://github.com/monocongo/climate_indices/commit/894bb8030b6c4582a567a510e91254e37344d299) (open in PR #1042; landed after #1026 merged)

### Findings and fix status
#### [P2] Pointwise reference mutated the session-scoped temperature fixture — Fixed
- **Location:** `tests/test_numerical_equivalence.py` (`_pointwise`, lines 49–58 at `894bb803`)
- **Origin:** Tests lane
- **Problem:** `_pointwise` passed a live `isel` view into `indices.pet`, and `eto.eto_thornthwaite` zeroes negative temperatures in place (`src/climate_indices/eto.py:343`). An independent probe showed 992 negative values in `gridded_monthly_temp_3d` before the Thornthwaite pass and 0 after, silently changing the session-scoped fixture and the Dask fixture derived from it.
- **Applied fix:** `_pointwise` hands each kernel a `.copy()`; verification probe confirms the fixture is byte-identical afterwards. Commit `046708ac`.

#### [P2] Docs claimed bit-for-bit SPEI while only gamma SPEI was enforced — Fixed
- **Location:** `docs/performance.md` (lines 97–104 at `894bb803`), `tests/test_numerical_equivalence.py:145` (`test_spei_*`)
- **Origin:** Docs lane, Contracts lane
- **Problem:** The policy listed SPEI unqualified in the bit-for-bit set, but `spei(..., distribution=Distribution.pearson)` routes through the same block L-moment fit the doc itself describes as ULP-prone, and the test hardcoded gamma.
- **Applied fix:** Policy scopes the bit-for-bit set to gamma-distributed SPI/SPEI and names the Pearson fit used by both indices as the ULP exception (`2fb62a99`); the SPEI test is parametrized over gamma (exact) and Pearson (`atol=1e-12`) so the claim is enforced (`2c128047`). Verified by `lane-docs-codex`.

#### [P2] Chunk-layout claim ignored the per-block Pearson gamma fallback — Fixed
- **Location:** `docs/performance.md` (lines 111–116 at `894bb803`)
- **Origin:** Tests lane
- **Problem:** The policy said chunk layout never changes SPI, but `spi` enables `fallback_to_gamma=True` and the fallback is applied per Spatial Block (`src/climate_indices/compute.py:1924-1945`), so a chunk shape that isolates different cells can change which cells fall back. The chunk test exercises the gamma path only.
- **Applied fix:** The paragraph now scopes chunk-layout bounds to gamma SPI/EDDI/percentage of normal plus PET and explains why Pearson is excluded. Commit `2fb62a99`.

#### [P3] All-NaN comparisons could pass vacuously — Fixed
- **Location:** `tests/test_numerical_equivalence.py` (`_pointwise`, line 57)
- **Origin:** Tests lane
- **Problem:** Every comparison uses `equal_nan=True`, so if a shared regression turned both paths fully NaN, every tight tolerance would assert nothing.
- **Applied fix:** `_pointwise` asserts `np.isfinite(expected).any()` before returning; partial vacuity is still caught by `equal_nan` mask checks. Commit `046708ac`.

#### [P3] Pearson ULP attribution overstated in docs and comments — Fixed
- **Location:** `docs/performance.md` (lines 99–101), `tests/test_numerical_equivalence.py:32-40`
- **Origin:** Correctness lane, Docs lane
- **Problem:** The drift was attributed specifically to the L-moment fit's scalar `math` vs vectorized NumPy transcendentals, but the block transform also evaluates `pearson3.cdf`/`norm.ppf` with broadcast parameters, and the fit was bit-identical on the reviewing host. The quoted 4.9e-15 could not be reproduced locally.
- **Applied fix:** Wording now attributes the drift to scalar-versus-broadcast evaluation of the same library call "on some CPUs", keeping the measured figures. Commit `2fb62a99`.

#### [P3] Two tolerance constants could drift from the docs — Fixed
- **Location:** `tests/test_numerical_equivalence.py` (`_ULP_ATOL`, line 40)
- **Origin:** Contracts lane
- **Problem:** `_THORNTHWAITE_ATOL` and `_PEARSON_ATOL` held the same value that `docs/performance.md` hardcodes as `atol=1e-12`; bumping one would silently falsify the docs.
- **Applied fix:** One `_ULP_ATOL` used by the Pearson parametrization and both PET assertions, with the per-path measurements kept in the comment. Commit `046708ac`.

#### [P3] Dask test did not prove laziness or fixture equality — Fixed
- **Location:** `tests/test_numerical_equivalence.py` (lines 262, 278, 293, 308, 316)
- **Origin:** Tests lane
- **Problem:** The test called `.compute()` on results without asserting the result was still a Dask array, so an eager materialization regression would make the chunk comparison a no-op. The two precipitation fixtures are generated independently from the same seed, so the "same field" premise was unpinned.
- **Applied fix:** Asserts the precip fixtures encode the same field and that each lazy result still has `.chunks` before computing. Commit `046708ac`.

#### [P3] Zero-probability branch had no exact-equivalence coverage — Fixed
- **Location:** `tests/test_numerical_equivalence.py:115` (`test_spi_zero_inflated_block_path_matches_the_serial_numpy_api`)
- **Origin:** Tests lane
- **Problem:** The new tight bounds ran only on zero-free, NaN-free input; the pre-existing `zero_inflated_precip_da` fixture was unused. A regression in the zero-count/`probabilities_of_zero` branch would only meet the looser `atol=1e-8` bounds in `tests/test_spatial_kernel.py`.
- **Applied fix:** A zero-inflated grid (two cells, ~50% zeros) is compared under the same gamma/Pearson bounds; verification confirmed non-zero `probabilities_of_zero` hits and no block-wide fallback. Commit `046708ac`.

#### [P3] SonarCloud duplication gate failed after the fix pass — Fixed
- **Location:** `tests/test_numerical_equivalence.py`
- **Origin:** Orchestrator (regression introduced by the zero-inflated fix above)
- **Problem:** The new zero-inflated test repeated the SPI test body, taking SonarCloud `new_duplicated_lines_density` to 12.2% against a 3% gate. The maintainer merged #1026 at `2c128047` before the dedup commit could be pushed, so the dedup missed the merge.
- **Applied fix:** Both SPI cases call a shared `_assert_spi_matches_the_serial_numpy_api` helper and the three parametrizations share `_ULP_CASES`; opened follow-up PR [#1042](https://github.com/monocongo/climate_indices/pull/1042) with commit `894bb803`, where SonarCloud reports 0.0% duplication and all CI checks pass.

#### [P4] "A Spatial Block is fitted as a unit" misread the fit scope — Fixed
- **Location:** `docs/performance.md:106`
- **Origin:** Docs lane
- **Problem:** The sentence read as one shared fit for the whole block, but normal Pearson/gamma parameters are fitted per `(period, cell)` inside one call; only the gamma fallback is block-wide.
- **Applied fix:** Replaced with "A Spatial Block runs one Pearson fit call over its cells" plus the fallback scope and ADR-0009 reference. Commit `2fb62a99`.

#### [P4] Dask/eager precipitation fixtures coupled by duplicated seed only — Fixed
- **Location:** `tests/test_numerical_equivalence.py:262`
- **Origin:** Tests lane
- **Problem:** `dask_monthly_precip_3d` and `gridded_monthly_precip_3d` regenerate the same field from independent `default_rng(99)` calls; editing one would break the test's premise with a confusing numeric diff. `tests/conftest.py` was left untouched because it is shared legacy scaffolding outside this PR's scope.
- **Applied fix:** The Dask test now asserts the two fixtures encode the same field before comparing derived results. Commit `046708ac`.

#### [P4] Unused `zero_inflated_precip_da` fixture — Fixed
- **Location:** `tests/conftest.py:590`, consumed at `tests/test_numerical_equivalence.py:115`
- **Origin:** Tests lane
- **Problem:** The fixture existed but no test consumed it.
- **Applied fix:** The new zero-inflated SPI case consumes it. Commit `046708ac`.

### Remaining blockers
None. #1026 is merged with all fixes except the SonarCloud dedup; that commit is in open follow-up PR [#1042](https://github.com/monocongo/climate_indices/pull/1042) and should be merged to land it.
